dropwizard-jdbi/src/main/java/io/dropwizard/jdbi/DBIFactory.java

```diff
@@ -53,7 +57,14 @@ public DBI build(Environment environment,
         final String validationQuery = configuration.getValidationQuery();
```

Comment:
This is the query that's used to validate connections in the pool (depending on settings, while idle, on borrow, on return), and the timeout wrapping is already implemented at the statement level in the Tomcat pool that DW uses.

Presuming that we trust the pool's validation activities, then just checking the age of `org.apache.tomcat.jdbc.pool.PooledConnection#getLastValidated` probably provides an adequate healthcheck (and that healthcheck already uses the timeout and lifecycle settings for the pool).